### PR TITLE
F gazebo namespaces - Fixes #75

### LINKED ISSUE #75: Hand simulator namespaces
Update the gazebo sim to use the same namespaces as the real hand. ie /rh

### PR DESCRIPTION
**Pull request by markpitchless**
Moves hand sim into /rh/.

I had to patch HandCommander to be able to take a namespace arg to run in a different namespace because it is used in the gazebo plugin via self_test, which needs to stay in the /gazebo namespace.

Also added a version of the test with a mock hand for those days when gazebo breaks.

### CODE REVIEW

File: sr_hand/launch/gazebo/gazebo_arm_and_hand.launch [hunk 1]
```diff
@@ -9,6 +9,7 @@
   <node name="spawn_support" pkg="gazebo_ros" type="spawn_model" args="-urdf -param support_description -z 0.0 -model support_model" respawn="false" output="screen" />
 
   <!-- spawn the arm and hand model -->
+  <group ns="rh">
```
Comment (ugocupcic) [inline sr_hand/launch/gazebo/gazebo_arm_and_hand.launch]: so we always have /rh in front of everything even when we only have one hand???

File: sr_hand/src/hand_commander.cpp [hunk 2]
```diff
@@ -54,7 +55,7 @@
       node_.param("robot_description", robot_desc_string, std::string());
       if (!robot_model.initString(robot_desc_string))
       {
-        ROS_ERROR("Couldn't parse the urdf file on sh_description or on robot_description.");
+        ROS_ERROR_STREAM("Couldn't parse the urdf file on sh_description or on robot_description (ns=" << node_.getNamespace() << ")");
```
Comment (ugocupcic) [inline sr_hand/src/hand_commander.cpp]: may be better to replace ns= with namespace= in the error message? Easier to understand.

File: sr_hand/launch/gazebo/gazebo_arm_and_hand.launch (hunk 1 above)
Comment (markpitchless) [inline sr_hand/launch/gazebo/gazebo_arm_and_hand.launch]: Damn, good point! Ok, so what I really want is to have a higher level launch for gazebo, like we have for the hand now (sr_ethercat_hand_config sr_rhand.launch etc), maybe rhand_gazebo.launch. Then this file can take the ns as an arg or put the group tag in the high level file but force /gazebo into the root here.

File: sr_hand/launch/gazebo/gazebo_arm_and_hand.launch [hunk 3]
```diff
@@ -1,7 +1,12 @@
 <launch>
+  <!-- Gazebo world file to load. -->
+  <arg name="world_name" default="$(find sr_description)/other/worlds/shadowhand.world" />
+  <!-- Namespace to run the hand under. -->
+  <arg name="hand_ns" default="rh" />
```
Comment (ugocupcic) [inline sr_hand/launch/gazebo/gazebo_arm_and_hand.launch]: what's that?

File: sr_hand/launch/gazebo/gazebo_hand.launch [hunk 4]
```diff
@@ -1,20 +1,35 @@
 <launch>
+  <!-- Gazebo world file to load. -->
+  <arg name="world_name" default="$(find sr_description)/other/worlds/shadowhand.world" />
+  <!-- Namespace to run the hand under. -->
+  <arg name="hand_ns" default="rh" />
```
Comment (ugocupcic) [inline sr_hand/launch/gazebo/gazebo_hand.launch]: same question: are you setting /rh/ as the namespace?

**Comment by toliver** [comment]
Ouch!!
Is this well well tested?
I was hoping I could finish the release thing before you merged this.

**Comment by toliver** [comment]
OK. I ran some tests on the finger rig, and it seems to work well. So going forward with the release.

**Pull request merged**